src/components/StyledList/StyledList.tsx

```diff
@@ -0,0 +1,65 @@
+import { Box, chakra, useMultiStyleConfig } from "@chakra-ui/react";
+import React, { forwardRef } from "react";
+
+import { checkListChildrenError } from "../List/List";
+import { TextSizes } from "../Text/Text";
+
+export interface StyledListProps {
+  /** Any child node passed to the component. */
+  children?: React.ReactNode;
+  /** A class name for the StyledList parent div. */
+  className?: string;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Data to render if `li` children elements are not passed. It must be an
+   * array of strings or JSX elements. */
+  listItems?: (string | JSX.Element)[];
+  /** The style used to render the StyledList component. For its initial release,
+   * there is only one variant style which is the default, "capped". */
+  style?: "capped";
+  /** The font size of the `li` elements. */
+  textSize?: TextSizes;
+}
+
+export const StyledList = chakra(
+  forwardRef<HTMLDivElement & HTMLUListElement, StyledListProps>(
+    (props, ref?) => {
+      const {
+        children,
+        className,
+        id,
+        listItems = [],
+        style = "capped",
+        textSize = "default",
+      } = props;
+      const styles = useMultiStyleConfig("StyledList", {
+        textSize,
+        variant: style,
+      });
+      let finalChildren;
```

Comment:
Just curious: why do you declare this variable here as opposed to just declaring it when you assign its value below?

Reply:
I didn't want the component to go through the assignment of `children` or run through the `map` function on line 54 if there were any issues. Those issues are caught before and the component just returns null -- extremely minor and negligible optimization. The js compiler does variable hoisting anyway so this pattern is similar to what it does.

Reply:
I also prefer declaring all variables at the beginning rather than in the middle of the component. Just a preference nitpicky thing though.